internal/provider/resources/deployment_schedule.go

```diff
@@ -225,6 +222,12 @@ func (r *DeploymentScheduleResource) Create(ctx context.Context, req resource.Cr
 		return
 	}
 
+	// The list of schedules returned by Create should only include the created schedule,
+	// not a list of all schedules on the deployment, so we can assume the first and only
+	// schedule in the list is the one we created.
+	//
+	// Additionally, we couldn't use getResourceByID here because of a race condition:
+	// we'd need an ID in the state to compare against, which doesn't exist yet.
```

Comment:
https://github.com/PrefectHQ/prefect/blob/93d59f35c36afa3281893fd648c08623f35a80a3/src/prefect/server/models/deployments.py#L835-L865

I realized the returned schedules here should only be the ones in the request, not all of the schedules currently on the deployment. This made it possible to support multiple schedules in the provider.